### PR TITLE
[Wip] pytest-tap to provide path to a test

### PR DESCRIPTION
**Pull request by Mattemagikern**
Ok, after some testing I think this is what I want to get as output, however I
haven't been able to verify that it actually works as intended on an actual
pytest test case.

Would you mind helping me as to how to test this plugin on live pytest?

(I'm not sure that it is correct, I just want the path to the testcase to be added to the printout) :P 

Signed-off-by: mattemagikern <mattemagikern@gmail.com>

### CODE REVIEW

**Comment by mblayman** [comment]
Sure, testing with the installed version can be done with the `tox` testing tool that is one of the included developer tools if you followed the Pipenv instructions. Tox will create a directory that contains virtualenv environments for different Python versions. After creating a virtualenv, it will install the package and run whatever commands are specified in the `tox.ini` file.

If I assume that you have Python 3.6 installed on your machine, you should be able to run:

```bash
$ tox -e py36
```

The tool will run the command listed in the `[testenv]` section. You can see in that file that `--tap-combined` is already listed. I think it will generate the output TAP file that you want to inspect.

**Comment by Mattemagikern** [comment]
I'm happy with the output now. What do you think?

It mimics the output provided by pytest, which in addition to the previous release adds the path to the test it runs in the tap output.

If this would be accepted into the master branch it would close #28

File: pytest_tap/tests/test_plugin.py [hunk 1]
```diff
@@ -72,15 +72,15 @@ def test_tracks_ok(self):
         report = mock.Mock(when='call', outcome='passed', location=location)
         plugin.pytest_runtest_logreport(report)
         plugin.tracker.add_ok.assert_called_once_with(
-            'test_file.py', 'TestFake.test_me')
+                'test_file.py', 'test_file.py::TestFake.test_me')
```
Comment (mblayman) [inline pytest_tap/tests/test_plugin.py]: Looks like there are extra spaces here. Could you update this so the indentation is only 4 spaces?

File: pytest_tap/tests/test_plugin.py [hunk 2]
```diff
@@ -135,4 +135,4 @@ def test_path_pytest(self):
         report = mock.Mock(when='call', outcome='passed', location=location)
         plugin.pytest_runtest_logreport(report)
         plugin.tracker.add_ok.assert_called_once_with(
-                'tests/test_file.py', 'TestFake.test_me')
+                'tests/test_file.py', 'tests/test_file.py::TestFake.test_me')
```
Comment (mblayman) [inline pytest_tap/tests/test_plugin.py]: Looks like there are extra spaces here. Could you update this so the indentation is only 4 spaces?

File: pytest_tap/plugin.py [hunk 3]
```diff
@@ -60,7 +60,7 @@ def pytest_runtest_logreport(report):
         report.when == 'call'
     ):
         return
-    description = str(report.location[2])
+    description = str(report.location[0]) + "::" + str(report.location[2])
```
Comment (mblayman) [inline pytest_tap/plugin.py]: Could you use single instead of double quotes?

**Comment by mblayman** [comment]
Hey @Mattemagikern, it looks like your editor might be set for 8 spaces or something. I see some changes that are unrelated to the functionality change. Could you please make sure that the indentation is only 4 spaces and undo the changes you made to other parts of the file? Also, the longer lengths have triggered a build failure since some lines are now too long.

File: pytest_tap/plugin.py [hunk 4]
```diff
@@ -23,10 +23,13 @@ def pytest_addoption(parser):
     group = parser.getgroup('terminal reporting', 'reporting', after='general')
     group.addoption(
         '--tap-stream', default=False, action='store_true', help=_(
-            'Stream TAP output instead of the default test runner output.'))
+            'Stream TAP output instead of the default test runner output.')
```
Comment (mblayman) [inline pytest_tap/plugin.py]: Please add the parenthesis back to the line above.

File: pytest_tap/plugin.py [hunk 5]
```diff
@@ -23,10 +23,13 @@ def pytest_addoption(parser):
...
+        )
     group.addoption(
         '--tap-files', default=False, action='store_true', help=_(
-            'Store all TAP test results into individual files per test case.'))
+            ('Store all TAP test results into individual files per test' +
```
Comment (mblayman) [inline pytest_tap/plugin.py]: Please revert to the previous style.

File: pytest_tap/plugin.py [hunk 6]
```diff
@@ -56,12 +59,12 @@ def pytest_configure(config):
 def pytest_runtest_logreport(report):
     """Add a test result to the tracker."""
     if not (
-        (report.when == 'setup' and report.outcome == 'skipped') or
-        report.when == 'call'
-    ):
+            (report.when == 'setup' and report.outcome == 'skipped') or
```
Comment (mblayman) [inline pytest_tap/plugin.py]: Please revert to the previous style.

File: pytest_tap/tests/test_plugin.py [hunk 7]
```diff
@@ -72,35 +72,36 @@ def test_tracks_ok(self):
         report = mock.Mock(when='call', outcome='passed', location=location)
         plugin.pytest_runtest_logreport(report)
         plugin.tracker.add_ok.assert_called_once_with(
-            'TestFake', 'TestFake.test_me')
+                'test_file.py', 'test_file.py::TestFake.test_me')
```
Comment (mblayman) [inline pytest_tap/tests/test_plugin.py]: Please remove 4 spaces from this line.

File: pytest_tap/tests/test_plugin.py [hunk 8]
```diff
@@ -72,35 +72,36 @@ def test_tracks_ok(self):
...
+            diagnostics='')
 
     def test_tracks_skip(self):
         plugin.tracker = mock.Mock()
         location = ('test_file.py', 1, 'TestFake.test_me')
         longrepr = ('', '', 'Skipped: a reason')
         report = mock.Mock(
-            when='setup', outcome='skipped', location=location,
-            longrepr=longrepr)
+                when='setup', outcome='skipped', location=location,
```
Comment (mblayman) [inline pytest_tap/tests/test_plugin.py]: Please remove 4 spaces from this line and the next one.

File: pytest_tap/tests/test_plugin.py [hunk 9]
```diff
@@ -72,35 +72,36 @@ def test_tracks_ok(self):
...
+                longrepr=longrepr)
         plugin.pytest_runtest_logreport(report)
         plugin.tracker.add_skip.assert_called_once_with(
-            'TestFake', 'TestFake.test_me', 'a reason')
+                'test_file.py', 'test_file.py::TestFake.test_me', 'a reason')
```
Comment (mblayman) [inline pytest_tap/tests/test_plugin.py]: Please remove 4 spaces from this line.

File: pytest_tap/tests/test_plugin.py [hunk 10]
```diff
@@ -72,35 +72,36 @@ def test_tracks_ok(self):
...
 
     def test_tracks_xfail(self):
         plugin.tracker = mock.Mock()
         location = ('test_file.py', 1, 'TestFake.test_me')
         report = mock.Mock(
-            when='call', outcome='skipped', location=location, wasxfail='')
+                when='call', outcome='skipped', location=location, wasxfail='')
```
Comment (mblayman) [inline pytest_tap/tests/test_plugin.py]: Please remove 4 spaces from this line.

File: pytest_tap/tests/test_plugin.py [hunk 11]
```diff
@@ -72,35 +72,36 @@ def test_tracks_ok(self):
...
         plugin.pytest_runtest_logreport(report)
         plugin.tracker.add_skip.assert_called_once_with(
-            'TestFake', 'TestFake.test_me', '')
+                'test_file.py', 'test_file.py::TestFake.test_me', '')
```
Comment (mblayman) [inline pytest_tap/tests/test_plugin.py]: Please remove 4 spaces from this line.

File: pytest_tap/tests/test_plugin.py [hunk 12]
```diff
@@ -128,3 +129,11 @@ def test_skips_reporting_with_no_output_option(self):
...
         plugin.pytest_unconfigure(config)
         self.assertFalse(plugin.tracker.generate_tap_reports.called)
+
+    def test_path_pytest(self):
+        plugin.tracker = mock.Mock()
+        location = ('tests/test_file.py', 1, 'TestFake.test_me')
+        report = mock.Mock(when='call', outcome='passed', location=location)
+        plugin.pytest_runtest_logreport(report)
+        plugin.tracker.add_ok.assert_called_once_with(
+                'tests/test_file.py', 'tests/test_file.py::TestFake.test_me')
```
Comment (mblayman) [inline pytest_tap/tests/test_plugin.py]: Please remove 4 spaces from this line.

**Comment by Mattemagikern** [comment]
Thank you for your guidance!

I will check it out!

Great!

I'll be watching the projects for things I can help out with! Great project! :)

**Pull request merged**